src/Workspaces/Remote/ServiceHub/Services/NavigateToSearch/RemoteNavigateToSearchService.cs

```diff
@@ -47,75 +51,90 @@ public ValueTask HydrateAsync(Checksum solutionChecksum, CancellationToken cance
+            return StreamWithSolutionAsync(solutionChecksum, SearchGeneratedDocumentsWorkerAsync, cancellationToken).WithJsonRpcSettings(
+                new JsonRpcEnumerableSettings { MaxReadAhead = MaxReadAhead });
+
+            async IAsyncEnumerable<RoslynNavigateToItem> SearchGeneratedDocumentsWorkerAsync(Solution solution, [EnumeratorCancellation] CancellationToken cancellationToken)
             {
                 var project = solution.GetRequiredProject(projectId);
-                var callback = GetCallback(callbackId, cancellationToken);
 
-                await AbstractNavigateToSearchService.SearchGeneratedDocumentsInCurrentProcessAsync(
-                    project, searchPattern, kinds.ToImmutableHashSet(), callback, cancellationToken).ConfigureAwait(false);
-            }, cancellationToken);
+                await foreach (var item in AbstractNavigateToSearchService.SearchGeneratedDocumentsInCurrentProcessAsync(
+                    project, searchPattern, kinds.ToImmutableHashSet(), cancellationToken).ConfigureAwait(false))
+                {
+                    yield return item;
+                }
+            }
         }
 
-        public ValueTask SearchCachedDocumentsAsync(ImmutableArray<DocumentKey> documentKeys, ImmutableArray<DocumentKey> priorityDocumentKeys, string searchPattern, ImmutableArray<string> kinds, RemoteServiceCallbackId callbackId, CancellationToken cancellationToken)
+        public IAsyncEnumerable<RoslynNavigateToItem> SearchCachedDocumentsAsync(
+            ImmutableArray<DocumentKey> documentKeys,
+            ImmutableArray<DocumentKey> priorityDocumentKeys,
+            string searchPattern,
+            ImmutableArray<string> kinds,
+            CancellationToken cancellationToken)
         {
-            return RunServiceAsync(async cancellationToken =>
-            {
-                // Intentionally do not call GetSolutionAsync here.  We do not want the cost of
-                // synchronizing the solution over to the remote side.  Instead, we just directly
-                // check whatever cached data we have from the previous vs session.
-                var callback = GetCallback(callbackId, cancellationToken);
-                var storageService = GetWorkspaceServices().GetPersistentStorageService();
-                await AbstractNavigateToSearchService.SearchCachedDocumentsInCurrentProcessAsync(
-                    storageService, documentKeys, priorityDocumentKeys, searchPattern, kinds.ToImmutableHashSet(), callback, cancellationToken).ConfigureAwait(false);
-            }, cancellationToken);
+            WorkspaceManager.SolutionAssetCache.UpdateLastActivityTime();
+
+            // Intentionally do not call GetSolutionAsync here.  We do not want the cost of
+            // synchronizing the solution over to the remote side.  Instead, we just directly
+            // check whatever cached data we have from the previous vs session.
```

Comment:
```suggestion
            // check whatever cached data we have from the previous VS session.
```

I recognize not new, but lowercase "vs" means "versus" to me.

Reply:
wfm :)

Reply:
done in next pr.